gp-includes/routes/translation-set.php

```diff
@@ -14,9 +14,13 @@
  */
 class GP_Route_Translation_Set extends GP_Route_Main {
 	public function new_get() {
-		$set             = new GP_Translation_Set();
-		$set->project_id = gp_get( 'project_id' );
-		$project         = $set->project_id ? GP::$project->get( $set->project_id ) : null;
+		$set        = new GP_Translation_Set();
+		$project_id = gp_get( 'project_id' );
+
+		// Make sure 'project_id' is a string and convert to int ID. Defaults to null.
+		$set->project_id = is_string( $project_id ) && intval( $project_id ) ? intval( $project_id ) : null;
```

Comment:
Same problem here. I have tested it with these GET parameters:

- `/glotpress/sets/-new/?project_id=fasdf`
- `/glotpress/sets/-new/?project_id=123`
- `/glotpress/sets/-new/?project_id=1s`
- `/glotpress/sets/-new/?project_id[]=1`
- `/glotpress/sets/-new/?project_id[1]=1ss`
- `/glotpress/sets/-new/?project_id=1234www`

```suggestion
		$set->project_id = filter_var( $project_id, FILTER_VALIDATE_INT ) !== false ? intval( $project_id ) : null;
```

Reply:
I'm making it more simpler with `is_numeric()`.